lib/gooddata/lcm/types/special/array.rb

```diff
@@ -0,0 +1,35 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2016 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative '../base_type'
+
+module GoodData
+  module LCM2
+    module Type
+      class ArrayType < BaseType
+        CATEGORY = :special
+
+        def initialize(type)
+          @type = type
+        end
+
+        def check(values)
+          return false unless values.kind_of?(Array)
+
+          values.each do |value|
+            return false unless @type.check(value)
+          end
+
+          true
+        end
+
+        def to_s
+          "#{self.class.name.split('::').last}<#{@type.to_s}>"
```

Comment:
could we use this one? https://github.com/gooddata/gooddata-ruby/pull/745/files#diff-189a95a928410eb41446e31c3bdea6cdR2